doc/develop/test/twister.rst

```diff
@@ -632,6 +632,26 @@ line break instead of white spaces.
 
 Most everyday users will run with no arguments.
 
+Test Statuses
+*************
+
+The following statuses are reported per testcase:
+
+- `passed`: A test is deemed to pass if its actual result matches its expected
+  result.
+- `failed`: A test is deemed to fail if its actual result does not match its
+  expected result.
+- `error`: A test produces the Error result when there is a problem in running
+  the test itself.
+- `skipped`: A test case that was skipped during a test script execution.
+- `blocked`: A test case that is scheduled to run but cannot run because of an
+  error or a crash in the suite or environment.
+- `notrun`: A test case or test suite that is not yet run.
```

Comment:
We've added info about `not run` tests in twister report, so I'm wondering when it can be reported as not 0?

Is it only in situations when we've interrupted twister execution?

Reply:
>If this continues to be confusing, we need to find some other terminilogy or a way to report those.

Maybe `built`? Indicating that the last stage was building and didn't went further. `built_only` is more verbose but maybe too lenghty?

Reply:
`built` works, `built_only` too verbose, and we have this info already in the reason field.
This will be used for the testsuite level, for the testcases under the same testsuite and if the build_only is not set in the yaml file, it will be marked as `blocked`, sounds good?

Reply:
I think using `blocked` can be problematic. Blocked is reserved for cases where some error occurred. From my understanding "blocked" is a "red" status alerting that something went wrong. I don't think it is the case here. If we need differentiation in statuses I guess "skip" would fit more. But why not just `built` regardless of `build_only` in yaml?